infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -142,9 +302,141 @@ private boolean registryTableHasNewEntries() {
         // Check if the log tail of registry table moved ahead
         if (currentLogTail != lastRegistryTableLogTail) {
             lastRegistryTableLogTail = currentLogTail;
+            registryTableEntries = runtime.getTableRegistry().getRegistryTable().entryStream().collect(Collectors.toList());
             return true;
         }
 
         return false;
     }
+
+    /**
+     * Preprocess the registration and logical group configuration tables, get the log tail that will be the start
+     * point for client config listener to monitor updates.
+     *
+     * @return Log tail hat will be the start point for client config listener to monitor updates.
+     */
+    public CorfuStoreMetadata.Timestamp preprocessAndGetTail() {
```

Comment:
The pattern of this method almost perfectly matches the `performFullSync()` right ? so can we consider using that resubscription pattern here as well?